client/src/main/java/org/asynchttpclient/netty/channel/DefaultChannelPool.java

```diff
@@ -148,25 +153,15 @@ private boolean isIdleTimeoutExpired(IdleChannel idleChannel, long now) {
             return idleTimeoutChannels != null ? idleTimeoutChannels : Collections.<IdleChannel> emptyList();
         }
 
-        private boolean isChannelCloseable(Channel channel) {
-            Object attribute = Channels.getAttribute(channel);
-            if (attribute instanceof NettyResponseFuture) {
-                NettyResponseFuture<?> future = (NettyResponseFuture<?>) attribute;
-                if (!future.isDone()) {
-                    LOGGER.error("Future not in appropriate state {}, not closing", future);
-                    return false;
-                }
-            }
-            return true;
-        }
-
         private final List<IdleChannel> closeChannels(List<IdleChannel> candidates) {
 
             // lazy create, only if we have a non-closeable channel
             List<IdleChannel> closedChannels = null;
             for (int i = 0; i < candidates.size(); i++) {
+                // We call takeOwnership here to avoid closing a channel that has just been taken out
+                // of the pool, otherwise we risk closing an active connection.
                 IdleChannel idleChannel = candidates.get(i);
-                if (isChannelCloseable(idleChannel.channel)) {
+                if (idleChannel.takeOwnership()) {
```

Comment:
So your opinion is that it's safe to drop `isChannelCloseable` because ownership guarantees that we can no longer never end up with a incomplete future?